### PR TITLE
8333962: Obsolete OldSize

### PR DESCRIPTION
**Pull request by albertnetymk**
Obsolete OldSize and related code. An internal variable `OldSize` is kept to capture the capacity of old-gen size.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8333962](https://bugs.openjdk.org/browse/JDK-8333962): Obsolete OldSize (**Enhancement** - P4)

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Zhengyu Gu](https://openjdk.org/census#zgu) (@zhengyu123 - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19647/head:pull/19647` \
`$ git checkout pull/19647`

Update a local copy of the PR: \
`$ git checkout pull/19647` \
`$ git pull https://git.openjdk.org/jdk.git pull/19647/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19647`

View PR using the GUI difftool: \
`$ git pr show -t 19647`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19647.diff">https://git.openjdk.org/jdk/pull/19647.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19647#issuecomment-2160101271)

### CODE REVIEW

File: src/hotspot/share/runtime/arguments.cpp [hunk 1]
```diff
@@ -540,6 +540,8 @@ static SpecialFlag const special_jvm_flags[] = {
   { "ParallelOldDeadWoodLimiterStdDev", JDK_Version::undefined(), JDK_Version::jdk(23), JDK_Version::jdk(24) },
   { "UseNeon",                      JDK_Version::undefined(), JDK_Version::jdk(23), JDK_Version::jdk(24) },
   { "ScavengeBeforeFullGC",         JDK_Version::undefined(), JDK_Version::jdk(23), JDK_Version::jdk(24) },
+
```
Comment (dholmes-ora) [inline src/hotspot/share/runtime/arguments.cpp:543]: Please remove blank line. but you need to sync with master to get recent updates to this table.

**Review by dholmes-ora — APPROVED** [review]
Al seems reasonable.

Thanks.

File: src/hotspot/share/runtime/arguments.cpp [hunk 2]
```diff
@@ -34,6 +34,7 @@
 #include "compiler/compilerDefinitions.hpp"
 #include "gc/shared/gcArguments.hpp"
 #include "gc/shared/gcConfig.hpp"
+#include "gc/shared/genArguments.hpp"
```
Comment (dholmes-ora) [inline src/hotspot/share/runtime/arguments.cpp:37]: Why is this needed?

**Review by dholmes-ora — APPROVED** [review]

File: src/hotspot/share/runtime/arguments.cpp (hunk 2 above)
Comment (albertnetymk) [reply src/hotspot/share/runtime/arguments.cpp:37]: `Arguments::set_heap_size` accesses `OldSize`, which is declared in this header.

Comment (dholmes-ora) [reply src/hotspot/share/runtime/arguments.cpp:37]: Got it. Thanks

File: src/hotspot/share/gc/shared/genArguments.hpp [hunk 3]
```diff
@@ -33,6 +33,8 @@ extern size_t MinNewSize;
 extern size_t MinOldSize;
 extern size_t MaxOldSize;
 
+extern size_t OldSize;
```
Comment (zhengyu123) [inline src/hotspot/share/gc/shared/genArguments.hpp:36]: Any reason we still want to keep `OldSize` variable? becase GCs really care about are `init`, `min` and `max` values.

Comment (albertnetymk) [reply src/hotspot/share/gc/shared/genArguments.hpp:36]: The concept of "initial old-gen size" will always be there. If `OldSize` is removed, all readers need to be updated to `InitialHeapSize - NewSize`. It's not obvious that is definitely better/more readable.

**Review by zhengyu123 — APPROVED** [review]
LGTM

**Comment by albertnetymk** [comment]
Synced with master. Running some tests before merging.

**Comment by albertnetymk** [comment]
Tests pass.

Thanks for review.

/integrate
